Source/Initialization/WarpXInitData.cpp

```diff
@@ -245,6 +245,96 @@ WarpX::PostProcessBaseGrids (BoxArray& ba0) const
         AMREX_D_TERM(},},})
         ba0 = BoxArray(std::move(bl));
     }
+
+    bool init_partition_species = false;
+    std::string density_function;
+    Real density_min=0;
+    Real efficiency_target = 0.7;
+    if (ParallelDescriptor::NProcs() > 1)
+    {
+        ParmParse pp_warpx("warpx");
+        std::string species;
+        pp_warpx.query("init_partition_species", species);
+        pp_warpx.query("init_partition_efficiency", efficiency_target);
+        if ( ! species.empty()) {
+            ParmParse pp(species);
+            std::string profile;
+            pp.query("profile", profile);
+            if (profile == "parse_density_function") {
```

Comment:
We should a warning here, for species that don't have a `parse_density_function`, to let the user know that they were ignored.